backend/services/monthly_dump_service.py

```
@@ -295,8 +296,25 @@ def _load_image(url: str) -> Optional[Image.Image]:
                 buffer.write(chunk)
 
             buffer.seek(0)
-            image = Image.open(buffer).convert("RGB")
-            return image
+            try:
+                # Open to inspect metadata without loading fully
+                image = Image.open(buffer)
+                
+                # Pixel limit
+                if image.width * image.height > MonthlyDumpService.MAX_IMAGE_PIXELS:
+                    logger.error(
```

Ruff (0.15.10)

[error] Too many arguments for `logging` format string

(PLE1205)